### ISSUE TITLE
Unexpected 'include_pdbs' option when compiling

### ISSUE THREAD

**Issue by rdeterre** [conan-io/conan#503]
Hi,

I'm not sure if here is the place for asking this. I am faced with an issue where a program that requires `gtest` fails with the following output during `conan install`:

```
ERROR: gtest/1.7.0@lasote/stable: 'options.include_pdbs' doesn't exist
'options' possible configurations are ['shared']
```

This error is surprising to me because the program does not define `include_pdbs`, nor does any of the other dependencies. 

In addition, the problem does not happen on all platforms. I have only seen it on a Red Hat 7 system running developer toolset 4.

Is there anything I can try to debug this problem further?

Thanks.

**Comment by memsharded** [comment]
Hi @rdeterre !

I am very surprised too. The options that can be defined in the gtest package are defined in the conanfile, can be seen there, and the only option available is that of ``shared``.

So the problem might come that someone in that machine is setting the options somewhere. The possible places are:

- The command line, something like ``$ conan install -o gtest:include_pdbs=True``
- A ``conanfile.txt`` file, in which an ``[options]`` section is defined, and in that section something like ``gtest:include_pdbs=xxx`` is written
- A previous ``conaninfo.txt`` file. That file is the result of an execution of the ``conan install`` command, and it caches the previously used values, so it is not necessary to type them again and again. That file will be located in the current folder, not in the ``conanfile.txt`` folder.

If you could verify that none of the above is happening. Otherwise, I have no idea, maybe having a look at the project, computer, a remote connection... we can chat if necessary to try to debug this issue. Thanks for reporting!

**Comment by rdeterre** [comment]
Thanks for your answer. I'm not sure if this can help, but I've found a way to make the problem go away here. The `conanfile.py` of the offending program had the following dependencies:

``` python
    requires = "libfwupdater/{}@alazar/stable".format(version),\
               "libprogressbar/1.1.0@alazar/stable",\
               "atsapi/5.10.15@alazar/testing"
```
but `libfwupdater` itself had a dependency on  `atsapi/5.10.15@alazar/testing`. When I removed `atsapi` from the `requires` variable, the problem went away.

Other than that, I had ensured that I was not calling the `include_pdb` option from the command line. I've pasted the contents of the `conanfile.py` of the offending program at the end in case this can help (after removal of the direct `atsapi` dependency). Lastly, I had removed all files from the build directory I was using to ensure that the option did not come from a local `conaninfo.txt` file. 

This was done on a RHEL 7 machine running the red hat developer toolset 4. I've deleted the `~/.conan` directory completely before starting, and had conan 0.12 installed.

I'm available to chat/call, I have access to the machine on which this was run during eastern time business hours.

Thanks again. 

``` python
from conans import ConanFile, CMake
import os

class FwupdaterCmdline(ConanFile):
    version = None
    version_major = None
    version_minor = None
    version_patch = None
    with open(os.path.join(
            os.path.dirname(os.path.realpath(__file__)),
            'version.txt')) as f:
        version = f.readline().rstrip()
        version_major = version.split('.')[0]
        version_minor = version.split('.')[1]
        version_patch = version.split('.')[2]
    print("Version is {}".format(version))

    requires = "libfwupdater/{}@alazar/stable".format(version),\
               "libprogressbar/1.1.0@alazar/stable"
    settings = "os", "compiler", "build_type", "arch"
    generators = "cmake"

    def build(self):
        cmake = CMake(self.settings)
        self.run('cmake "{}" {} -DVERSION_MAJOR={} -DVERSION_MINOR={} '
                 '-DVERSION_PATCH={}'.format(
            self.conanfile_directory,
            cmake.command_line,
            self.version_major,
            self.version_minor,
            self.version_patch))
        self.run('cmake --build . %s' % cmake.build_config)

    def imports(self):
        self.copy("*.dll", dst="bin", src="bin")
        self.copy("*.so*", dst="bin", src="bin")
```

**Comment by memsharded** [comment]
I still can't figure out what is happening. Having dependencies like A->B->C & A->C is fine, nothing should fail. Furthermore, it still surprise me the ``include_pdbs``, I have absolutely no idea where it can come from. Is it an option of some of those packages you are depending? I have checked the conan.io, but I guess those are private packages in your server.

I am going to add some tests with the A->B->C & A->C configuration, just in case, but if you could find/grep for ``include_pdbs`` in all your package recipes, that could help. Thanks again!

**Comment by memsharded** [comment]
I think I got it. I have done some test for A->B->C & A->C, and they are not the problem, but the way some options are being handled in such layout. Specifically the issue appears in the GTest package deleting its own ``include_pdbs`` option. I will try to come up with a fix tomorrow, just wanted to let you know, thanks for your help!

**Comment by rdeterre** [comment]
Awesome, thanks for building this great project, and thanks for your such quick support!

**Comment by memsharded** [comment]
You are welcome, thanks to you for using it and giving feedback!

The issue has been fixed in https://github.com/conan-io/conan/pull/507, but it wont be available until next release (unless you run it from source). I have also submitted a local fix PR to Gtest package: https://github.com/lasote/conan-gtest/pull/11, so if it gets merged, you might be able to try it immediately.

**Comment by memsharded** [comment]
The PR in gtest https://github.com/lasote/conan-gtest/pull/11 has been merged. You can now try your original setup with your dependencies, please tell me if worked. Thanks!

**Comment by rdeterre** [comment]
The problem is indeed gone here, thanks!

**Comment by memsharded** [comment]
Released in conan 0.13.1
